src/libsyntax/parse/parser.rs

```diff
@@ -2856,7 +2856,7 @@ impl<'a> Parser<'a> {
             }
 
             // Otherwise, we use the unique pointer default.
-            let subexpression = self.parse_prefix_expr();
+            let subexpression = self.parse_prefix_expr(prefix_prec);
```

Comment:
So this means that `box ..5` does not parse, which I consider a bug. I'm not really clear on why we change from `min_prec` to `prefix_prec` here.